### ISSUE TITLE
Issue while building code

### ISSUE THREAD

**Issue by dharmjit** [IBM-Blockchain/learn-chaincode#130]
Hi, I am new to hyperledger fabric and while building using go build . I am getting below error
```
# github.com/hyperledger/fabric/vendor/github.com/tecbot/gorocksdb
exec: "gcc": executable file not found in %PATH%
# github.com/hyperledger/fabric/vendor/github.com/mattn/go-sqlite3
exec: "gcc": executable file not found in %PATH%
```

**Comment by dhyey20** [comment]
What env are you running this on?
If it is ubuntu try installing build-essentials. 

Also post `go version` output.

**Comment by dharmjit** [comment]
its windows 7. I guess I need to install some gcc compiler. I will try that and update here. 
go version go1.7.4 windows/amd64

**Comment by dharmjit** [comment]
After installing tdm-gcc-5.1.0-3.exe and go version 1.6 I am getting below error.
`# github.com/hyperledger/fabric/vendor/github.com/tecbot/gorocksdb
..\vendor\github.com\tecbot\gorocksdb\backup.go:4:24: fatal error: rocksdb/c.h: No such file or directory
 // #include "rocksdb/c.h"`

**Comment by masterDev1985** [comment]
@dharmjit could you also please confirm that you are on the correct branch of the fabric and have pulled all the latest changes?

```
$ cd $GOPATH/src/github.com/hyperledger/fabric
$ git status
On branch v0.6

$ git pull https://github.com/hyperledger/fabric.git v0.6
From https://github.com/hyperledger/fabric
 * branch              v0.6       -> FETCH_HEAD
Already up-to-date.
```

Also, you are only running `go build` on the learn-chaincode tutorial files, correct?  You are not trying to run the command against the fabric repository?

**Comment by dharmjit** [comment]
Ohh I am very sorry. I was running this against fabric repo. Can I also use docker setup for the same? Thanks for helping me out.

**Comment by masterDev1985** [comment]
No worries!  If you are trying to build the fabric from source, then you'll need to follow instructions from hyperledger for setting up a development environment:

http://hyperledger-fabric.readthedocs.io/en/latest/dev-setup/build.html

For docker, you might want to keep your eye on the IBM Blockchain Dockerhub account: 

https://hub.docker.com/r/ibmblockchain/fabric-peer/

There are some published images for fabric v0.6, but none for v1.0 just yet.  There are `make` commands in the fabric repositories that should build docker images for 1.0, if you need them.  I'd get access to a Linux box before going down this route.  Also, this repository only teaches you how to write chaincode for v0.6, not v1.0.  Here are some examples of v1.0 chaincode directly from the fabric:

https://github.com/hyperledger/fabric/tree/master/examples/chaincode/go
